code/game/objects/structures/noticeboard.dm

```diff
@@ -19,67 +19,106 @@
 
 //attaching papers!!
 /obj/structure/noticeboard/attackby(obj/item/weapon/O, mob/user, params)
-	if(istype(O, /obj/item/weapon/paper))
+	if(istype(O, /obj/item/weapon/paper) || istype(O, /obj/item/weapon/photo))
+		if(!allowed(user))
+			user << "<span class='info'>You are not authorized to add notices</span>"
+			return
 		if(notices < 5)
-			if(!user.drop_item())
+			if(!user.unEquip(O))
 				return
-			O.add_fingerprint(user)
-			add_fingerprint(user)
 			O.loc = src
 			notices++
-			icon_state = "nboard0[notices]"	//update sprite
-			user << "<span class='notice'>You pin the paper to the noticeboard.</span>"
+			icon_state = "nboard0[notices]"
+			user << "<span class='notice'>You pin the [O] to the noticeboard.</span>"
 		else
-			user << "<span class='notice'>You reach to pin your paper to the board but hesitate. You are certain your paper will not be seen among the many others already attached.</span>"
+			user << "<span class='notice'>The notice board is full</span>"
 	else
 		return ..()
 
 /obj/structure/noticeboard/attack_hand(mob/user)
-	var/dat = "<B>Noticeboard</B><BR>"
-	for(var/obj/item/weapon/paper/P in src)
-		dat += "<A href='?src=\ref[src];read=\ref[P]'>[P.name]</A> <A href='?src=\ref[src];write=\ref[P]'>Write</A> <A href='?src=\ref[src];remove=\ref[P]'>Remove</A><BR>"
+	var/auth = allowed(user)
+	var/dat = "<B>[name]</B><BR>"
+	for(var/obj/item/P in src)
+		if(istype(P, /obj/item/weapon/paper))
+			dat += "<A href='?src=\ref[src];read=\ref[P]'>[P.name]</A> [auth ? "<A href='?src=\ref[src];write=\ref[P]'>Write</A> <A href='?src=\ref[src];remove=\ref[P]'>Remove</A>" : ""]<BR>"
+		else
+			dat += "<A href='?src=\ref[src];read=\ref[P]'>[P.name]</A> [auth ? "<A href='?src=\ref[src];remove=\ref[P]'>Remove</A>" : ""]<BR>"
 	user << browse("<HEAD><TITLE>Notices</TITLE></HEAD>[dat]","window=noticeboard")
 	onclose(user, "noticeboard")
 
-
 /obj/structure/noticeboard/Topic(href, href_list)
 	..()
 	usr.set_machine(src)
 	if(href_list["remove"])
 		if((usr.stat || usr.restrained()))	//For when a player is handcuffed while they have the notice window open
 			return
-		var/obj/item/P = locate(href_list["remove"])
-		if((P && P.loc == src))
-			P.loc = get_turf(src)	//dump paper on the floor because you're a clumsy fuck
-			P.add_fingerprint(usr)
-			add_fingerprint(usr)
+		var/obj/item/I = locate(href_list["remove"])

```

Comment:
You should do locate in contents to prevent them locating any item in the world, so they have to locate the item in the noticeboards contents.

The rest of the lines below probably protects anyway, but it will be safer if someone comes along and meddles with your topic call later.